xarray/core/parallel.py

```diff
@@ -456,13 +457,21 @@ def subset_dataset_to_block(
                     for dim in variable.dims
                 }
                 subset = variable.isel(subsetter)
+                if name in chunk_index:
```

Comment:
TODO: There's a more general version of this condition that looks for unnecessary broadcasting and minimizes it.